agent/internal/client/timeout_sender_test.go

```diff
@@ -58,21 +64,45 @@ func TestLogSenders(t *testing.T) {
 	suite.Run(t, &s)
 }
 
-func (s *logSenderSuite) SetupTest() {
-	s.underlyingSenders = []send.Sender{}
-}
-
 func (s *logSenderSuite) SetupSuite() {
-	s.restClient = NewHostCommunicator("foo", "hostID", "hostSecret").(*hostCommunicator)
+	s.server, _ = newMockServer(func(w http.ResponseWriter, _ *http.Request) {
```

Comment:
I'm having trouble following: why do we need this server now?

Reply:
Because the default logger requires it (it needs to fetch the task to setup the taskoutput sender).